### PR TITLE
[Synthetics] Adding basicAuth param + table fix

### PR DESCRIPTION
**Pull request by l0k0ms**
### What does this PR do?

* Adds the `basicAuth` parameter documentation for API and browser tests
* Fixes the table layout to be pure markdown.
* Re-order the arguments section in order to clearly show master attributes and child attributes
* Document assertions parameter for the SSL test
* Document the request parameters (`host` and `port`) for the SSL test
* Updates the Postman collection accordingly
* Fixes Start/pause synthetics test API doc thanks to https://github.com/DataDog/documentation/issues/5473

### Motivation

PM request

### Preview link

* https://docs-staging.datadoghq.com/gus/synthetic/api/?lang=python#create-a-test
* https://docs-staging.datadoghq.com/gus/synthetic/api/?lang=python#edit-a-test

### CODE REVIEW

File: content/en/api/synthetics/post_test.md [hunk 1]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+*   **`port`** - _required SSL test_ - If creating a SSL test, specify the port to connect to.
+*   **`basicAuth`** - _optional_ - If your endpoint is behind basic Auth, use this parameter to define your username and password with the following value: `{"username": "<USERNAME>", "password": "<PASSWORD>"}`.
+*   **`timeout`** - _optional_ - When the request should timeout.
+*   **`headers`** - _optional API test_ - Headers in the API request.
+*   **`body`** - _optional API test_ - The body for the API request. Accepts text strings (including a JSON as a text string). Specify the type using `Content-Type` `property` parameter andthe type (like `application/json` or `text/plain` in the `headers` paramater.
+*   **`cookies`** - _optional API test_ - Cookies to send along with your API test request.
+
+#### Options
+
+The **`options`** argument is required for the browser, API and SSL tests. Use it to specify custom request headers, authentication credentials, body content, cookies, or to have the test follow redirects. All optional parameters take their default value if you don't specify a value. It's a JSON object with the following attributes available:
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
The **`options`** argument is required for the browser, API, and SSL tests. Use it to specify custom request headers, authentication credentials, body content, cookies, or have the test follow redirects. All optional parameters take their default value if you don't specify a value. It's a JSON object with the following attributes available:
```

File: content/en/api/synthetics/post_test.md [hunk 2]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+
+*  **`tick_every`:** - _required_ -  How often the test should run (in seconds - current possible values are 60, 300, 900, 1800, 3600, 21600, 43200, 86400, 604800).
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
*  **`tick_every`:** - _required_ -  How often the test should run (in seconds - current possible values are 60, 300, 900, 1800, 3600, 21600, 43200, 86400, and 604800).
```

File: content/en/api/synthetics/post_test.md [hunk 3]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+*  **`min_failure_duration`** - _optional_ - How long the test should be in failure before alerting (integer, number of seconds, max 7200). Default is 0.
+*  **`min_location_failed`** - _optional_ - The minimum number of locations that must have been in failure at the same time during at least one moment in the `min_failure_duration` period (min_location_failed and min_failure_duration are part of the advanced alerting rules  - integer, >= 1. Default is 1.
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
*  **`min_location_failed`** - _optional_ - The minimum number of locations in failure at the same time during at least one moment in the `min_failure_duration` period (min_location_failed and min_failure_duration are part of the advanced alerting rules  - integer, >= 1). Default is 1.
```

File: content/en/api/synthetics/post_test.md [hunk 4]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+*  **`follow_redirects`** - _optional_ - boolean - whether to follow redirects or not (max ten redirects can be followed before triggering a "Too many redirects" error). Valid values are `true` or `false`. Default value is `false`.
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
*  **`follow_redirects`** - _optional_ - boolean - whether to follow redirects (a max of ten redirects can be followed before triggering a "Too many redirects" error). Valid values are `true` or `false`. Default value is `false`.
```

File: content/en/api/synthetics/post_test.md [hunk 5]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+*  **`device_ids`** - _required browser test_ - The type of device you want to use to test. To get a list of available devices, use the `GET devices for browser checkes` method and use the `id` from the response. At least one device is required.
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
*  **`device_ids`** - _required browser test_ - The type of device used to test. To get a list of available devices, use the `id` from the response of the `GET devices for browser checks` method. At least one device is required.
```

File: content/en/api/synthetics/post_test.md [hunk 6]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+
+#### Assertions
+
+The **`assertions`**  argument is required for the API and SSL tests. It allows you to define exactly what should happen for a test to be considered passed. **It's an array of JSON objects** with the following attributes available:
+
+*   **`type`** - _required_ - The part of the response that you want to assess. Possible types are:
+
+  * `header`: When you define a header, you must specify the header parameter key in the `property` parameter, and the header parameter value with the `target` parameter.
+  * `body`: Use the `target` attribute to specify the expected value for `body`.
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
  * `body`: Use the `target` attribute to specify the expected value for the `body`.
```

File: content/en/api/synthetics/post_test.md [hunk 7]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+  * `responseTime`: Use the `target` attribute to specify the expected value for `responseTime`. For example: `"target":180000`.
+  * `statusCode`: Use the `target` attribute to specify the expected value for `statusCode`. For example: `"target":403`.
+  * `certificate`:  For SSL test only, use the `target` attribute to specify the expected value for `certificate`. `"target":1` is a valid certificate, `"target":0` is a non-valid certificate.
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
  * `certificate`:  For SSL tests only, use the `target` attribute to specify the expected value for `certificate`. `"target":1` is a valid certificate, `"target":0` is a non-valid certificate.
```

File: content/en/api/synthetics/post_test.md [hunk 8]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+  * `property`: For SSL test only, when you define a certificate `property`, you must specify the certificate property key in the `property` parameter, and the certificate property value with the `target`parameter.
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
  * `property`: For SSL tests only, when you define a certificate `property`, you must specify the certificate property key in the `property` parameter, and the certificate property value with the `target`parameter.
```

File: content/en/api/synthetics/post_test.md [hunk 9]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+
+*   **`target`** - _required_ - The expected value for the assertion.
+*   **`property`** - _optional_ - When you are setting up a `header` for the `type` parameter, this is required to define the headers parameter key. Valid values are any header keys, like `Content-Type` or `Authorization`.
+*   **`operator`** - _required_ - Defines how to compare the target and the actual value from the response. Valid operators depend on the `type` of assertions. This is the list of valid operators per type:
+
+| `type`        | Valid `operator`                                                            | Value type                                                                                                                                           |
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
| type        | Valid operator                                                            | Value type                                                                                                                                           |
```

File: content/en/api/synthetics/post_test.md [hunk 10]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+| ---           | ---                                                                         | ---                                                                                                                                                  |
+| `header`       | `contains`, `does not contain`, `is`, `is not`, `matches`, `does not match` | For `contains`/`does not contain`/`is`/`is not`: String for `matches`/`does not match`: [RegexString][3] |
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
| `header`       | `contains`, `does not contain`, `is`, `is not`, `matches`, `does not match` | String for `contains`/`does not contain`/`is`/`is not`.  [RegexString][3] for: `matches`/`does not match`. |
```

File: content/en/api/synthetics/post_test.md [hunk 11]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+| `body`          | `contains`, `does not contain`, `is`, `is not`, `matches`, `does not match` | For `contains`/`does not contain`/`is`/`is not`: String for `matches`/`does not match`: [RegexString][3] |
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
| `body`          | `contains`, `does not contain`, `is`, `is not`, `matches`, `does not match` | String for: `contains`/`does not contain`/`is`/`is not`. [RegexString][3] for: `matches`/`does not match`. |
```

File: content/en/api/synthetics/post_test.md [hunk 12]
```diff
@@ -11,59 +11,71 @@ Create a Synthetics test to initiate and configure the tests you want Datadog to
...
+| `responseTime` | `lessThan`                                                                  | Integer                                                                                                                                              |
+| `statusCode`   | `is`, `is not`                                                              | Integer                                                                                                                                              |
+| `certificate` | `isInMoreThan` | Integer |
+| `property` | `contains`, `does not contain`, `is`, `is not`, `matches`, `does not match` | For `contains`/`does not contain`/`is`/`is not`: String for `matches`/`does not match`: [RegexString][3] |
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/post_test.md]:
```suggestion
| `property` | `contains`, `does not contain`, `is`, `is not`, `matches`, `does not match` | String for: `contains`/`does not contain`/`is`/`is not`. [RegexString][3] for: `matches`/`does not match`. |
```

File: content/en/api/synthetics/put_test.md [hunk 13]
```diff
@@ -15,59 +15,72 @@ A browser test is treated like a GET API test. This method gives you the ability
...
-   *   **`method`** - _required_ - The method of the API to test. Valid values are `GET`, `POST`, `PUT`, `PATCH`, and `DELETE`. Use `GET` for a browser test.
-   *   **`url`** - _required_ - The endpoint for the API Datadog is testing. For a browser test, the URL for the website Datadog is testing.
-   *   **`timeout`** - _optional_ - When the API request will timeout.
-   *   **`headers`** - _optional_ - Headers in the API request.
-   *   **`body`** - _optional_ The body for the API request. Accepts text strings (including a JSON as a text string). Specify the type using `Content-Type` `property` parameter and the type (like `application/json` or `text/plain` in the `headers` paramater.
-*   **`locations`** - _required_ - A list of the locations that you want the tests to be sent from. At least one value is required, and you can use all locations. For a list of valid locations, use the `GET available locations` method. At least one value is required, and you can use all locations.
-*   **`message`** - _required_ - A description of the test.
+### Arguments
+
+The main arguments for your test are:
```
Comment (ruthnaebeck) [inline content/en/api/synthetics/put_test.md]: Do similar edits for this file.

**Comment by MargotLepizzera** [comment]
LGTM 👍

File: content/en/api/synthetics/put_test.md (hunk 13 above)
Comment (l0k0ms) [reply content/en/api/synthetics/put_test.md]: Done, and pushed to the postman collection file.

**Pull request merged**